src/midend/mir_lower.cpp

```diff
@@ -41,7 +42,8 @@ namespace graphit {
         //  sets the flags for other parts of the lowering process
         ApplyExprLower(mir_context, schedule).lower();
 
-
+        // TODO more context after getting it to work
```

Comment:
Add some comments for this additional lowering pass?

Reply:
Yep!